service/index.js

```diff
@@ -240,6 +265,40 @@ module.exports = class AuthenticationService {
       if (!deletedCount && (strict === true)) return Promise.reject(new Error(`Failed to remove user: ${email}.`));
     });
   }
+
+  /**
+   * Authorizes third party then logs in or signs up the user.
+   */
+  authorizeThirdParty(code, provider) {
+    if (!code || !provider) {
+      return Promise.reject(
+        new Error('Required parameters missing for third party authorization.')
+      );
+    }
+
+    return p(this).deferrari.deferUntil(THIRD_PARTY_CONFIG)
+    .then(authieThirdParty => authieThirdParty.login(code, provider)
+      .then(result => p(this).deferrari.deferUntil(CONNECTED)
+        .then(models => models.AuthUser.findOne({where: {email: result.email}})
```

Comment:
Makes sense, but this could happen if the user tries to authorize through third-party being already logged in with a local account. Most probably, the application won't know either user id or email unless the user is logged in. The use-case for this that I can picture right now is that if there is an option to explicitly link an existing account to any one of these third party accounts. If this presumption is correct then that scenario probably should be handled separately. What do you think.